### PR TITLE
JBIDE-19091 BrowserSim: Fixing LiveReload. Refactoring

### PR DESCRIPTION
**Pull request by ibuziuk**

### CODE REVIEW

File: plugins/org.jboss.tools.vpe.browsersim/src/org/jboss/tools/vpe/browsersim/ui/BrowserSim.java [hunk 1]
```diff
@@ -566,6 +570,7 @@ public void reinitSkin() {
 		skin.getBrowser().setUserAgent(device.getUserAgent());
 		skin.getBrowser().setUrl(oldSkinUrl); 
 		skin.getShell().open();
+		processLiveReload(specificPreferences.isEnableLiveReload());
```
Comment (maxandersen) [inline plugins/org.jboss.tools.vpe.browsersim/src/org/jboss/tools/vpe/browsersim/ui/BrowserSim.java]: what does this do/fix ?

File: plugins/org.jboss.tools.vpe.browsersim/src/org/jboss/tools/vpe/browsersim/ui/BrowserSim.java [hunk 2]
```diff
@@ -400,46 +400,50 @@ public void run() {
...
-				+ 	"};" //$NON-NLS-1$
-
-				// Overriding window.onerror 
-				+	"window.onerror = function(msg, url, lineNumber) {" //$NON-NLS-1$
-				+		"console.error(msg + ' on line ' + lineNumber + ' for ' + url);"  //$NON-NLS-1$
-				+	"}" //$NON-NLS-1$
-				+ "}" //$NON-NLS-1$
-		+ "})();");	 //$NON-NLS-1$
+	private void overrideJsLogFunctions(final IBrowser browser) {
+		if (browser != null && !browser.isDisposed()) {
```
Comment (maxandersen) [inline plugins/org.jboss.tools.vpe.browsersim/src/org/jboss/tools/vpe/browsersim/ui/BrowserSim.java]: +1 for guarding the call but I wonder *why* is it this method is even called when browser is null ? is there a listener that gets triggered that should have been removed instead ?

File: plugins/org.jboss.tools.vpe.browsersim/src/org/jboss/tools/vpe/browsersim/ui/BrowserSim.java (hunk 1 above)
Comment (ibuziuk) [inline plugins/org.jboss.tools.vpe.browsersim/src/org/jboss/tools/vpe/browsersim/ui/BrowserSim.java]: Basically, this is the fix for livereload. it enables it, if needed, if the 'Refresh' button from the 'Platforms' toolbar was pressed.

File: plugins/org.jboss.tools.vpe.browsersim/src/org/jboss/tools/vpe/browsersim/ui/BrowserSim.java (hunk 2 above)
Comment (ibuziuk) [inline plugins/org.jboss.tools.vpe.browsersim/src/org/jboss/tools/vpe/browsersim/ui/BrowserSim.java]: @maxandersen frankly, I do not think it is called when the browser is null or disposed in **this** case, due to the fact that there were no NPE or widget disposed for console.log functions. Adding this checks has nothing in common with the original JBIDE-19091 issue (I just did it while refactoring). As I recall removing listeners is unnecessary - they are removed when a control is disposed automatically. However, I do believe you are right about the PR for CordovaSim RippleInjector where the "widget disposed" was caught - https://github.com/jbosstools/jbosstools-browsersim/pull/51

Comment (maxandersen) [inline plugins/org.jboss.tools.vpe.browsersim/src/org/jboss/tools/vpe/browsersim/ui/BrowserSim.java]: okey - makes sense.

**Comment by ibuziuk** [comment]
released to master

**Pull request closed without merging**